packages/core-data/src/selectors.js

```diff
@@ -732,12 +732,7 @@ export function __experimentalGetTemplateForLink( state, link ) {
 
 	const template = records?.length ? records[ 0 ] : null;
 	if ( template ) {
-		return getEditedEntityRecord(
-			state,
-			'postType',
-			'wp_template',
-			template.id
-		);
+		return getEntityRecord( state, 'postType', 'wp_template', template.id );
```

Comment:
I think this is a very recent change by @jorgefilipecosta that was needed for the title editing in template mode. My suggestion there was that this selector should probably just return the "id" of the template and use smaller selectors later to fetch the things we need like "title"...

Reply:
Yes we need the edited entity otherwise changes to the title are not reflected and we will cause a regression. Returning an id and having small selectors sounds like the way to go.
Regarding getEditedEntityRecord it seems unexpected that the selector returns a new object if there were no changes. We should probably check if we can improve the selector.

Reply:
I fixed the selectors instead.